python/loyalty_workflow.py

```diff
@@ -0,0 +1,135 @@
+import logging
+from datetime import timedelta
+
+from temporalio import activity, workflow
+from temporalio.common import WorkflowIDReusePolicy
+from temporalio.exceptions import WorkflowAlreadyStartedError
+from temporalio.workflow import ParentClosePolicy
+from temporalio.client import WorkflowHandle
+
+import email_strings
+from shared import *
+
+
+# Basic activity that logs and does string concatenation
+@activity.defn(name="SendEmail")
+async def send_email(body: str):
+    activity.logger.info("Sending email with contents %s" % body)
+
+
+# Basic workflow that logs and invokes an activity
+@workflow.defn
+class CustomerLoyaltyWorkflow:
+    def __init__(self):
+        self.customer = None
```

Comment:
```suggestion
        self.customer: Optional[Customer] = None
```

I think you should add MyPy type checking and use annotations everywhere you can